File: common_library/src/main/java/com/phone/common_library/manager/TextViewStyleManager.java
package com.phone.common_library.manager;

import android.content.Context;
import android.graphics.Typeface;
import android.text.Spannable;
import android.text.SpannableString;
import android.text.style.AbsoluteSizeSpan;
import android.text.style.StyleSpan;
import android.widget.TextView;

import com.phone.common_library.spannable.VerticalAlignTextSpan;

public class TextViewStyleManager {

    private static final String TAG = TextViewStyleManager.class.getSimpleName();

    public static void setTextViewStyle(TextView textView,
                                        String data,
                                        int start) {
        Spannable spannable = new SpannableString(data);
        // 粗体
        spannable.setSpan(new StyleSpan(Typeface.BOLD), start, data.length(), Spannable.SPAN_INCLUSIVE_INCLUSIVE);
        // 显示
        textView.setText(spannable);
    }

    public static void setTextViewStyle(Context context,
                                        TextView textView,
                                        String data,
                                        int start,
                                        int end,
                                        float size) {
        Spannable spannable = new SpannableString(data);
        // 粗体
        spannable.setSpan(new StyleSpan(Typeface.BOLD), start, end, Spannable.SPAN_INCLUSIVE_INCLUSIVE);
        spannable.setSpan(new AbsoluteSizeSpan(ScreenManager.spToPx(context, size)), start, end, Spannable.SPAN_INCLUSIVE_INCLUSIVE);
        // 显示
        textView.setText(spannable);
    }

    public static void setTextViewStyleVerticalCenter(Context context,
                                        TextView textView,
                                        String data,
                                        int start,
                                        int end,
                                        float size) {
        Spannable spannable = new SpannableString(data);
        // 粗体
        spannable.setSpan(new StyleSpan(Typeface.BOLD), start, end, Spannable.SPAN_INCLUSIVE_INCLUSIVE);
        spannable.setSpan(new AbsoluteSizeSpan(ScreenManager.spToPx(context, size)), start, end, Spannable.SPAN_INCLUSIVE_INCLUSIVE);
        //SpannableString字体大小不一致垂直居中显示
        spannable.setSpan(new VerticalAlignTextSpan(context, size), start, end, SpannableString.SPAN_INCLUSIVE_EXCLUSIVE);
        // 显示
        textView.setText(spannable);
    }

}

File: library_common/src/main/java/com/phone/library_common/manager/TextViewStyleManager.kt
package com.phone.library_common.manager

import android.graphics.Typeface
import android.text.Spannable
import android.text.SpannableString
import android.text.style.AbsoluteSizeSpan
import android.text.style.StyleSpan
import android.widget.TextView
import com.phone.library_common.manager.ScreenManager.spToPx
import com.phone.library_common.spannable.VerticalAlignTextSpan

object TextViewStyleManager {

    private val TAG = TextViewStyleManager::class.java.simpleName

    @JvmStatic
    fun setTextViewStyle(
        textView: TextView,
        data: String,
        start: Int
    ) {
        val spannable: Spannable = SpannableString(data)
        // 粗体
        spannable.setSpan(
            StyleSpan(Typeface.BOLD),
            start,
            data.length,
            Spannable.SPAN_INCLUSIVE_INCLUSIVE
        )
        // 显示
        textView.text = spannable
    }

    @JvmStatic
    fun setTextViewStyle(
        textView: TextView,
        data: String?,
        start: Int,
        end: Int,
        size: Float
    ) {
        val spannable: Spannable = SpannableString(data)
        // 粗体
        spannable.setSpan(StyleSpan(Typeface.BOLD), start, end, Spannable.SPAN_INCLUSIVE_INCLUSIVE)
        spannable.setSpan(
            AbsoluteSizeSpan(spToPx(size)),
            start,
            end,
            Spannable.SPAN_INCLUSIVE_INCLUSIVE
        )
        // 显示
        textView.text = spannable
    }

    @JvmStatic
    fun setTextViewStyleVerticalCenter(
        textView: TextView,
        data: String?,
        start: Int,
        end: Int,
        size: Float
    ) {
        val spannable: Spannable = SpannableString(data)
        // 粗体
        spannable.setSpan(StyleSpan(Typeface.BOLD), start, end, Spannable.SPAN_INCLUSIVE_INCLUSIVE)
        spannable.setSpan(
            AbsoluteSizeSpan(spToPx(size)),
            start,
            end.toInt(),
            Spannable.SPAN_INCLUSIVE_INCLUSIVE
        )
        //SpannableString字体大小不一致垂直居中显示
        spannable.setSpan(
            VerticalAlignTextSpan(size),
            start,
            end,
            SpannableString.SPAN_INCLUSIVE_INCLUSIVE
        )
        // 显示
        textView.text = spannable
    }

}